libfreerdp/core/input.c

```diff
@@ -967,6 +997,19 @@ BOOL freerdp_input_send_extended_mouse_event(rdpInput* input, UINT16 flags, UINT
 	if (freerdp_settings_get_bool(input->context->settings, FreeRDP_SuspendInput))
 		return TRUE;
 
+	/* Save cursor position in prevent-session-lock mode */
+	if (input->context->settings->FakeMouseMotionInterval)
+	{
+		struct timespec ts;
+		timespec_get(&ts, TIME_UTC);
+		if (flags & (PTR_FLAGS_MOVE | PTR_FLAGS_BUTTON1 | PTR_FLAGS_BUTTON2 | PTR_FLAGS_BUTTON3))
```

Comment:
* use function for updates
* adjust the flags, extended mouse event has no mouse wheel events but only click events